src/core/thread/mesh_forwarder.cpp

```diff
@@ -573,12 +574,32 @@ void MeshForwarder::SetRxOnWhenIdle(bool aRxOnWhenIdle)
 
 void MeshForwarder::SetPollPeriod(uint32_t aPeriod)
 {
-    if (mMac.GetRxOnWhenIdle() == false && mPollPeriod != aPeriod)
+    if (!mNetif.IsUp())
```

Comment:
Thanks a lot for this explanation and proposed enhancements about this, Jonathan. I see what you mean. May I have one question here, if other components maintain their own timer respectively and when timer fires, the component will send out data poll request at once. Yes, it's indeed no need to call SetPollPeriod() first, and no competition as well. But which component and when will call SetPollPeriod() to start a timer with default period value in this case? Currently only MLE component calls this function as SED child.